php/libraries/LorisForm.class.inc

```diff
@@ -0,0 +1,486 @@
+<?php
+/**
+ * This class represents an attempt to re-implement the API for
+ * HTML_QuickForm (Version 1) in Loris since QuickForm1 isn't
+ * maintained anymore, and QuickForm2 is a completely different API.
+ *
+ * It only reimplements the minimal set required for Loris to work,
+ * and doesn't do it in a particularly elegant way, but prevents
+ * PHP warnings that QuickForm1 generates since QuickForm1 was written
+ * for PHP4.
+ *
+ * PHP Version 5
+ *
+ * @category Main
+ * @package  Main
+ * @author   Dave MacFarlane <david.macfarlane2@mcgill.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+
+/**
+ * Reimplementation of QuickForm API from scratch
+ *
+ * @category Main
+ * @package  Main
+ * @author   Dave MacFarlane <david.macfarlane2@mcgill.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class LorisForm
+{
+    var $form          = array();
+    var $defaultValues = array();
+
+    /**
+     * Adds an element to the $this->form array. Handles the creation
+     * of any shared attributes that are shared between different
+     * element types.
+     *
+     * @param string $name    The element name.
+     * @param string $label   The label to attach to this element.
+     * @param array  $attribs An array of other attributes that should
+     *                        get added. Currently only the "class"
+     *                        attribute gets added.
+     *
+     * @return &array A reference to the array that was added to $this->form
+     */
+    function &addBase($name, $label, $attribs)
```

Comment:
If the function returns a reference.

Call-time pass-by-reference is deprecated in PHP, not define-time pass-by-reference.